dagshub/data_engine/model/datasource.py

```diff
@@ -1200,6 +1210,80 @@ def contains(self, item: str):
         self._test_not_comparing_other_ds(item)
         return self.add_query_op("contains", item)
 
+    def _periodic_filter(self, periodtype, items):
+        periods = [str(s) for s in items]
+
+        if not self._query.timezone:
+            self._query.timezone = _get_local_timezone()
+
+        return self.add_query_op(periodtype, periods)
+
+    def date_field_in_years(self, *item: int):
+        """
+        Checks if a metadata field (which is of datetime type) is in one of given years list.
+        local timezone assumed unless time_zone() requests anything else.
+
+        Args:
+            List of years.
+
+        Examples::
+
+            datasource[(datasource["y"].date_field_in_years(1979, 2003)
+
+        """
+
+        return self._periodic_filter("year", item)
+
+    def date_field_in_months(self, *item: int):
+        """
+        Checks if a metadata field (which is of datetime type) is in one of given months list.
+        local timezone assumed unless time_zone() requests anything else.
+
+        Args:
+            List of months.
+
+        Examples::
+
+            datasource[(datasource["y"].date_field_in_months(12, 2)
+
+        """
+        return self._periodic_filter("month", item)
+
+    def date_field_in_days(self, *item: int):
+        """
+        Checks if a metadata field (which is of datetime type) is in one of given days list.
+        local timezone assumed unless time_zone() requests anything else.
+
+        Args:
+            List of days.
+
+        Examples::
+
+            datasource[(datasource["y"].date_field_in_days(25, 2)
+
+        """
+        return self._periodic_filter("day", item)
+
+    def date_field_in_timeofday(self, item: str):
+        """
+        Checks if a metadata field (which is of datetime type) is in given minute range inside the day (any day).
+        range is in the format of: "HH:mm-HH:mm" where start hour is on the left.
+        a range that starts at one day and ends at next day,
+        should be expressed as OR of 2 range filter.
+        local timezone assumed unless time_zone() requests anything else.
+
+        Args:
+            Time range string.
+
+        Examples::
+
+            datasource[(datasource["y"].date_field_in_timeofday("11:30-12:30")
+
+        """
+        self._test_not_comparing_other_ds(item)
+        self._query.timezone = _get_local_timezone()
```

Comment:
oops

Reply:
if not self._query.timezone:
            self._query.timezone = _get_local_timezone()

Reply:
Oops 🙃 
Make a PR to fix it I guess 🤷